Content.Server/Botany/Systems/MutationSystem.cs

```diff
@@ -200,6 +214,63 @@ private void MutateHarvestType(ref HarvestType val, int bits, int totalbits, flo
             val = HarvestType.SelfHarvest;
     }
 
+    private void MutateGases(ref Dictionary<Shared.Atmos.Gas, float> gases, float min, float max, int bits, int totalbits, float mult)
+    {
+        float p = mult * bits / totalbits;
+        p = Math.Clamp(p, 0, 1);
+        if (!Random(p))
+            return;
+
+        List<Shared.Atmos.Gas> newGas = Enum.GetValues(typeof(Shared.Atmos.Gas)).Cast<Shared.Atmos.Gas>().ToList();
+        var rng = IoCManager.Resolve<IRobustRandom>();
```

Comment:
This probably should be a dependency

Reply:
also the gases cast looks extremely sus

Reply:
I could have listed all of the gases by hand but that at least makes sure they all are accounted for in case more ever are added.

Reply:
Made it part of PostInject instead.

Reply:
1. i feel theres already a thing for getting a random gas see gas leak event
2. if not then `_random.Pick(Enum.GetValues(typeof(Gas)));` also `using Content.Shared.Atmos;` bruh

Reply:
1. predefined whitelist.
2. random.pick takes `IReadOnlyList` not `Array`. The  `using Content.Shared.Atmos;` is fair point.

Reply:
Array implements IList and thus IReadOnlyList no?

Reply:
Nope!